### PR TITLE
T41424 Enable autoindex in nginx

### PR DESCRIPTION
**Pull request by JenySadadia**
Fixes https://github.com/kernelci/kernelci-api/issues/230 

Created directory `config`  under `docker.storage` to store configuration file for `storage` service.
Added `nginx.conf` file to modify `server` block copied from `/etc/nginx/conf.d/default.conf` to enable autoindex.
Mounted the config file in `docker-compose`.

### CODE REVIEW

**Comment by gctucker** [comment]
Tested OK on staging: http://staging.kernelci.org:9080/

**Comment by gctucker** [comment]
Note: Browsers might try to automatically redirect to https://staging.kernelci.org:9080/ which won't work as there's no SSL certificate set up for this.  You can work around this by opening it in an incognito window or by adding an exception in the browser settings.

**Comment by gctucker** [comment]
(also `curl http://staging.kernelci.org:9080/` works...)

**Comment by gctucker** [comment]
I've moved the API storage directory to `/data/storage.staging.kernelci.org` on the server and bind-mounted it so it's now available both via port 9080 and also here (which works with HTTPS):
https://storage.staging.kernelci.org/api/

**Comment by gctucker** [comment]
@mgalka @VinceHillier 

I've added this line in `fstab` so it will mount it again upon reboot:
```
/data/storage.staging.kernelci.org/api		/home/kernelci/kernelci-deploy/checkout/kernelci-api/docker/storage/data	none	bind
```
Please note that `README.txt` is under source control, so if that gets changed in the `storage.kernelci.org` directory it will appear in the checkout.  So it shouldn't get garbage-collected.

**Review by gctucker — APPROVED** [review]

File: docker/storage/config/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    #
+    #location ~ \.php$ {
+    #    root           html;
+    #    fastcgi_pass   127.0.0.1:9000;
+    #    fastcgi_index  index.php;
+    #    fastcgi_param  SCRIPT_FILENAME  /scripts$fastcgi_script_name;
+    #    include        fastcgi_params;
+    #}
+
+    # deny access to .htaccess files, if Apache's document root
```
Comment (gctucker) [inline docker/storage/config/nginx.conf:39]: So actually, we could maybe just have added `.htaccess` with the `autoindex` option if Nginx supports it like Apache.  But a config file is probably cleaner anyway.

**Comment by JenySadadia** [comment]
Thanks for the testing @gctucker 
Adding `server` block within `http` context to the default `/etc/nginx/nginx.conf` file was another way of enabling `autoindex`.
However, the config file for nginx is subject to the change with different nginx version.
Hence, I decided to add custom configuration to `/etc/nginx/conf.d/default.conf` file.

File: docker/storage/config/nginx.conf
Comment (gctucker) [inline docker/storage/config/nginx.conf:1]: Ah I missed the fact this was being copied as `default.conf`.  Maybe this file should be renamed to `default.conf` to remove ambiguity as it's not actually `nginx.conf`?

Comment (gctucker) [reply docker/storage/config/nginx.conf:1]: Maybe just make another commit / PR to fix it with a rename?

Comment (JenySadadia) [reply docker/storage/config/nginx.conf:1]: Created https://github.com/kernelci/kernelci-api/pull/233

**Pull request merged**